rotkehlchen/chain/ethereum/modules/airdrops/decoder.py

```diff
@@ -176,6 +203,53 @@ def _decode_fpis_claim(  # pylint: disable=no-self-use
 
         return None, None
 
+    def _decode_elfi_claim(  # pylint: disable=no-self-use
+            self,
+            tx_log: EthereumTxReceiptLog,
+            transaction: EthereumTransaction,  # pylint: disable=unused-argument
+            decoded_events: List[HistoryBaseEntry],  # pylint: disable=unused-argument
+            all_logs: List[EthereumTxReceiptLog],  # pylint: disable=unused-argument
+            action_items: List[ActionItem],  # pylint: disable=unused-argument
+    ) -> Tuple[Optional[HistoryBaseEntry], Optional[ActionItem]]:
+        """Example:
+        https://etherscan.io/tx/0x1e58aed1baf70b57e6e3e880e1890e7fe607fddc94d62986c38fe70e483e594b
+        """
+        if tx_log.topics[0] != ELFI_VOTE_CHANGE:
+            return None, None
+
+        user_address = hex_or_bytes_to_address(tx_log.topics[1])
+        delegate_address = hex_or_bytes_to_address(tx_log.topics[2])
+        raw_amount = hex_or_bytes_to_int(tx_log.data[0:32])
+        amount = asset_normalized_value(amount=raw_amount, asset=A_ELFI)
+
+        # now we need to find the transfer, but can't use decoded events
+        # since the transfer is from one of at least 2 airdrop contracts to
+        # vote/locking contract. Since neither the from, nor the to is a
+        # tracked address there won't be a decoded transfer. So we search for
+        # the raw log
+        for other_log in all_logs:
+            if other_log.topics[0] != ERC20_OR_ERC721_TRANSFER:
+                continue
+
+            transfer_raw = hex_or_bytes_to_int(other_log.data[0:32])
+            if other_log.address == A_ELFI.ethereum_address and transfer_raw == raw_amount:
+                event = HistoryBaseEntry(
+                    event_identifier=transaction.tx_hash.hex(),
+                    sequence_index=self.base.get_sequence_index(tx_log),
+                    timestamp=ts_sec_to_ms(transaction.timestamp),
+                    location=Location.BLOCKCHAIN,
+                    location_label=user_address,
+                    asset=A_ELFI,
+                    balance=Balance(amount=amount),
+                    notes=f'Claim {amount} ELFI from element-finance airdrop and delegate it to {delegate_address}',  # noqa: E501
```

Comment:
When you self delegate it will be shown here as your address but would be better to not show it?

Reply:
Wanted to avoid extra conditionals .. but well since you ask for it and I need to push to remove commented code will add it :)